libr/util/skiplist.c

```diff
@@ -198,6 +198,15 @@ R_API RSkipListNode* r_skiplist_insert(RSkipList* list, void* data) {
 	return x;
 }
 
+R_API bool r_skiplist_insert_autofree(RSkipList* list, void* data) {
+	RSkipListNode* node = r_skiplist_insert (list, data);
+	if (node && data != node->data) { // duplicate
+		free (data);
```

Comment:
uhm maybe this should be calling the list.free function pointer instead of just free

Reply:
right